File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadChart3D/ViewModel/UserDataViewModel.cs
using OpenSilver.Samples.TelerikUI.RadChart3D.Helpers;
using System.Collections.Generic;

namespace OpenSilver.Samples.TelerikUI.RadChart3D.ViewModel
{
    public sealed class UserDataViewModel
    {
        private IList<IEnumerable<double>> _data;
        private int _itemsCount;
        private int _seriesCount;

        public IList<IEnumerable<double>> CollectionData
        {
            get
            {
                if (_data == null)
                {
                    _data = FillSampleChartData(SeriesCount, ItemsCount);
                }

                return _data;
            }
        }

        public IEnumerable<double> Data
        {
            get
            {
                return CollectionData[0];
            }
        }

        public int ItemsCount
        {
            get
            {
                return _itemsCount;
            }
            set
            {
                _itemsCount = value;
            }
        }

        public int SeriesCount
        {
            get
            {
                return _seriesCount;
            }
            set
            {
                _seriesCount = value;
            }
        }

        private IList<IEnumerable<double>> FillSampleChartData(int seriesCount, int numbOfItems)
        {
            List<IEnumerable<double>> itemsSource = new List<IEnumerable<double>>();

            for (int i = 0; i < seriesCount; i++)
            {
                itemsSource.Add(SeriesExtensions.GetUserData(numbOfItems, i));
            }

            return itemsSource;
        }
    }
}

File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadChart3D/ViewModel/UserDataViewModel.vb
Imports OpenSilver.Samples.TelerikUI.RadChart3D.Helpers
Imports System.Collections.Generic

Namespace OpenSilver.Samples.TelerikUI.RadChart3D.ViewModel
    Public NotInheritable Class UserDataViewModel
        Private _data As IList(Of IEnumerable(Of Double))
        Private _itemsCount As Integer
        Private _seriesCount As Integer

        Public ReadOnly Property CollectionData As IList(Of IEnumerable(Of Double))
            Get
                If _data Is Nothing Then
                    _data = FillSampleChartData(SeriesCount, ItemsCount)
                End If

                Return _data
            End Get
        End Property

        Public ReadOnly Property Data As IEnumerable(Of Double)
            Get
                Return CollectionData(0)
            End Get
        End Property

        Public Property ItemsCount As Integer
            Get
                Return _itemsCount
            End Get
            Set(value As Integer)
                _itemsCount = value
            End Set
        End Property

        Public Property SeriesCount As Integer
            Get
                Return _seriesCount
            End Get
            Set(value As Integer)
                _seriesCount = value
            End Set
        End Property

        Private Function FillSampleChartData(seriesCount As Integer, numbOfItems As Integer) As IList(Of IEnumerable(Of Double))
            Dim itemsSource As List(Of IEnumerable(Of Double)) = New List(Of IEnumerable(Of Double))()

            For i = 0 To seriesCount - 1
                itemsSource.Add(RadChart3D.Helpers.SeriesExtensions.GetUserData(numbOfItems, i))
            Next

            Return itemsSource
        End Function
    End Class
End Namespace
